docs/declarative-customization/view-formatting.md

```diff
@@ -154,13 +155,306 @@ This example uses the `rowFormatter` element, which totally overrides the render
   }
 }
 ```
+
 You can find this sample with additional details here: [Multi-line view rendering](https://github.com/SharePoint/sp-dev-list-formatting/tree/master/view-samples/multi-line-view)
 
+### Build custom group headers and footers
+
+You can use `groupProps` to format group headers with flexibility to add grouped column's data, display name and item count. You can also add group aggregates in the group headers or format it directly in the group footers.
+
+#### Example: Color coded group header
+
+In the example below we have list with group headers formatted according to column metadata.
+![Employee list grouped by City with formatted group header](../images/employee-formatted-group-header-list-layout.png)
+

```

Comment:
We should write some commentary about the JSON code here to make user aware - just like above other JSON snippets like use of groupProps and header formatter